tasks/validate-isin/src/lib.rs

```diff
@@ -0,0 +1,49 @@
+extern crate luhn_test_of_credit_card_numbers;
+
+use luhn_test_of_credit_card_numbers::luhn_test;
+
+fn validate_isin(isin: &str) -> bool {
+    if !isin.chars().all(|x| x.is_alphanumeric()) || isin.len() != 12 {
+        return false;
+    }
+    if !isin[..2].chars().all(|x| x.is_alphabetic())
+        || !isin[2..12].chars().all(|x| x.is_alphanumeric())
+        || !isin.chars().last().unwrap().is_numeric()
+    {
+        return false;
+    }
+
+    let bytes = isin.as_bytes();
+
+    let s2 = bytes
+        .iter()
+        .flat_map(|&c| {
+            if c.is_ascii_digit() {
+                vec![c]
+            } else {
+                (c + 10 - ('A' as u8)).to_string().into_bytes()
+            }
+        })
+        .collect::<Vec<u8>>();
+
+    let string = std::str::from_utf8(&s2).unwrap();
+    let number = string.parse::<usize>().unwrap();
```

Comment:
The problem probably lies here. `usize` will be `u32` on 32-bit, which is likely too small to hold the number. Try parsing a `u64` instead.

Reply:
Nice catch, thanks! It works now :)